### DISCUSSION TITLE
Filter-Mask breaks with many andRelatedTo

### DISCUSSION THREAD

**Discussion by simon-eichenberger** [craftcms/cms#13458] [General]
I need to build a Filter-Mask that allows users to select checkboxes for categories from 3 groups. Each category group has about 30 cateogries. I have about 5000 entries and my current approach of making a query with andRelatedTo leads to a breakdown of the server when i have about 6-7 relations.

```
$finder = Entry		
foreach($checkedCategoryIds as $categoryId) {
	$finder->andRelatedTo(['targetElement' => $categoryId]);
}::find();
$finder->section($section);
$finder->limit($limit);
$finder->offset($offset);
$finder->orderBy('dateCreated DESC');
$finder->all();
```

Is there sth wrong with my query or has somebody an advide of handling multiple and relations in a performant way?

Any help appreaciated.

**Comment by brandonkelly** [comment]
Seems like that could be simplified to a single `relatedTo()` call pretty easily:

```php
$finder = Entry::find();
$finder->relatedTo(['targetElement' => $checkedCategoryIds]);
$finder->section($section);
// ...
```
